File: java/src/main/java/workshop/encryption/exercise3/Application.java
package workshop.encryption.exercise3;

import workshop.encryption.exercise4.SharingUser;

public class Application {

    public static SecureSession SESSION;

    public void login(String username, String password) throws Exception {
        SharingUser user = Users.get(username);
        String contentKey = user.getContentKey(password);
        
        SESSION = new SecureSession(user, contentKey);
    }
}

File: ruby/lib/exercise3/application.rb
class Application

  def login(username, password)
    user = Users.get(username)
    content_key = user.content_key(password)
    private_key = user.private_key(password)
    @@session = SecureSession.new(user, content_key, private_key)
  end

  def self.session
    @@session
  end

end
